devito/finite_differences/elementary.py

```diff
@@ -0,0 +1,805 @@
+import sympy
+
+from devito.finite_differences.differentiable import (DifferentiableFunction, diffify)
+
```

Comment:
ahem... I guess we need `__all__ = [...]` or we'll end up exposing DifferentiableFunction, diffify, and sympy too, to the user API 😬

Reply:
`__all__` is nothing so nothing is exposed. Need to be imported explicitly.

Reply:
wait, what? IIRC, this is only true if a variable name begins with an `_`  -- then it's hidden away. Otherwise it's implicitly imported, unless `__all__` is present. [Here](https://docs.python.org/3/tutorial/modules.html#importing-from-a-package) seems to confirm this

Reply:
assuming I'm correct (if not apologies in advance), note that albeit tedious, it's not a shame to have a long list of explicit exports, see for example [here](https://github.com/sympy/sympy/blob/master/sympy/__init__.py#L51)

Reply:
From your link
```
If __all__ is not defined, the statement from sound.effects import * does not import all submodules from the package sound.effects into the current namespace; it only ensures that the package sound.effects has been imported
``